command/kv_put.go

```diff
@@ -0,0 +1,249 @@
+                          specified without -modify-index, the key will first be
+                          fetched and the resulting ModifyIndex will be used on
+                          the next query. The default value is false.
+
+  -flags=<int>            Unsigned integer value to assign to this key-value
+                          pair. This value is not read by Consul, so clients can
+                          use this value however makes sense for their use case.
+                          The default value is 0 (no flags).
+
+  -modify-index=<int>     Unsigned integer representing the ModifyIndex of the
+                          key. This is often combined with the -cas flag, but it
+                          can be specified for any key. The default value is 0.
+
+  -release                Forfeit the lock on the key at the givne path. This
+                          requires the -session flag to be set. The key must be
+                          held by the session in order to be unlocked. The
+                          default value is false.
+
+  -session=<string>       User-defined identifer for this session as a string.
+                          This is commonly used with the -acquire and -release
+                          operations to build robust locking, but it can be set
+                          on any key. The default value is empty (no session).
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (c *KVPutCommand) Run(args []string) int {
+	cmdFlags := flag.NewFlagSet("get", flag.ContinueOnError)
+	cmdFlags.Usage = func() { c.Ui.Output(c.Help()) }
+	httpAddr := HTTPAddrFlag(cmdFlags)
+	datacenter := cmdFlags.String("datacenter", "", "")
+	token := cmdFlags.String("token", "", "")
+	stale := cmdFlags.Bool("stale", false, "")
+	cas := cmdFlags.Bool("cas", false, "")
+	flags := cmdFlags.Uint64("flags", 0, "")
+	modifyIndex := cmdFlags.Uint64("modify-index", 0, "")
+	session := cmdFlags.String("session", "", "")
+	acquire := cmdFlags.Bool("acquire", false, "")
+	release := cmdFlags.Bool("release", false, "")
+	if err := cmdFlags.Parse(args); err != nil {
+		return 1
+	}
+
+	// Check for arg validation
+	args = cmdFlags.Args()
+	key, data, err := c.dataFromArgs(args)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error! %s", err))
+		return 1
+	}
+
+	if (*release || *acquire) && *session == "" {
+		c.Ui.Error("Error! Missing -session (required with -acquire and -release)")
+		return 1
+	}
+
+	// Create and test the HTTP client
+	conf := api.DefaultConfig()
+	conf.Address = *httpAddr
+	conf.Token = *token
+	client, err := api.NewClient(conf)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error connecting to Consul agent: %s", err))
+		return 1
+	}
+
+	pair := &api.KVPair{
+		Key:         key,
+		ModifyIndex: *modifyIndex,
+		Flags:       *flags,
+		Value:       []byte(data),
+		Session:     *session,
+	}
+
+	wo := &api.WriteOptions{
+		Datacenter: *datacenter,
+		Token:      *token,
+	}
+
+	switch {
```

Comment:
Hmm. I'm not super familiar with that stuff. Did you want me to do something here or were ya just thinking out loud?